crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1585,7 +1547,13 @@ impl<'src> Parser<'src> {
         self.parse_comma_separated_list(RecoveryContextKind::DictElements, |parser| {
             if parser.eat(TokenKind::DoubleStar) {
                 keys.push(None);
-                values.push(parser.parse_expression_with_bitwise_or_precedence().expr);
+
+                // Handle dictionary unpacking. Here, the grammar is `'**' bitwise_or`
+                // which requires limiting the expression.
+                let value = parser.parse_conditional_expression_or_higher();
+                parser.validate_expression_with_bitwise_or_precedence(&value);
+
```

Comment:
Would it make sense to keep a `parse_expression_with_bitwise_or_precedence` that calls into `parse_conditional_expression_or_higher` and calls into the validation to avoid the repetition of the validation code?

Reply:
Yeah, I like that